pyne/mcnp.py

```diff
@@ -1399,13 +1399,25 @@ def mat_from_inp_line(filename, mat_line, densities='None'):
     # collect all material card data on one string
     line_index = 1
     line = linecache.getline(filename, mat_line + line_index)
+    lib_id = {}
     # people sometimes put comments in materials and then this loop breaks                                                                                       # so we need to keep reading if we encounter comments
     while len(line.split()) > 0 and (line[0:5] == '     ' or line[0].lower() == 'c'):
+        
+        # For deafult cross-section library. Requires a seperate line(s) of library indicator
+        # in material card 
+        lib_name = ['nlib', 'plib', 'hlib', 'pnlib', 'elib']
+        if line.split()[0][0:4] in lib_name:
```

Comment:
I wonder if you could just loop through the entries in `line.split()` to see if any of them match?  This would allow the library indicator to be at any position, and also allow there to be multiple library indicators per line.

Reply:
This modification works for any combination of multiple library indicators (including all in a single line) as long as any library indicator begins in a separate line or lines. I also tried to extend this modification through a mixture of library indicator and isotope indicator along with a fraction indicator but I did not achieve yet. As I understood from the function of `mat_from_inp_line`, searching for all possible cases needs a deep change in the considered function due to the way how it was written. But still there may be a short-cut. I will be working on it ! Besides the reader ability, I am going to add the writer capability for default library in this PR.